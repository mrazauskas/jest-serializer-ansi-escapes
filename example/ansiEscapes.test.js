const { expect, jest, test } = require("@jest/globals");
const ansiEscapesSerializer = require("jest-serializer-ansi-escapes");

expect.addSnapshotSerializer(ansiEscapesSerializer);

const output = [];

jest
  .spyOn(process.stdout, "write")
  .mockImplementation((chunk) => output.push(chunk));

test("ansi escapes", () => {
  process.stdout.write("\u001b[1;2mLoading...\u001b[0m");
  process.stdout.write("\u001b[2K\u001b[G");
  process.stdout.write("\u001b[3;32mSuccess!\u001b[0m");

  expect(output.join("")).toMatchSnapshot();
});
