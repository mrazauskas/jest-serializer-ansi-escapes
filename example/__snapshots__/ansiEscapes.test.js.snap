// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ansi escapes 1`] = `
"<boldDim>Loading...</>
<eraseLine>
<moveCursorToColumn1>
<italicGreen>Success!</>"
`;
